src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -70,6 +70,10 @@ ov::Tensor Bank::get(int64_t uid, const std::string& device) {
     // uid may be coming from a 2nd (3rd, ...) model
     // detach the tensor here just in case
     const_cast<LazyTensor&>(iter_device->second.lt).detach();
+    // Also detach from registered tensors
+    auto it_registered = device_bank.registered_tensors.find(iter_device->second.lt);
+    NPUW_ASSERT(it_registered != device_bank.registered_tensors.end());
+    const_cast<LazyTensor&>(it_registered->first).detach();
```

Comment:
Previously our `.get()` had effects. If the requested `LazyTensor` hasn't been evaluated yet, it is evaluated and copied to the device memory. And we did `.detach()` right after.

After the recent changes, `.get()` is just `.get()`. So here's the question - if there's no allocation happens at this point, why what has to be detached hasn't been detached at this point yet?

It seems we have too many `.detaches()` now - it just means our flow is not very clear.

Reply:
Done